README.md

```diff
@@ -20,22 +29,113 @@ Or install it yourself as:
 
 ## Usage
 
-This gem is still under development. The code below is how I plan to make this gem work. It does not work yet.
-
-```Ruby
+```ruby
 require 'octoprint'
 
-Octoprint.configure(host: 'https://octopi.local/', api_key: 'j98G2nsJq...')
+# Configure globally for single printer
+Octoprint.configure(host: 'https://octopi.local/', api_key: 'your_api_key')
+
+# List files with full type safety  
+files = Octoprint::Files.list                    # => Array<Octoprint::File>
+files.first.name                                 # => String (typed)
+files.first.type_path                           # => Array<String> (typed)
 
-# list files
-Octoprint::Files.list
+# Get current job with progress tracking
+job = Octoprint::Job.current                     # => Octoprint::Job  
+puts "#{job.file.name}: #{job.progress.percent}%" if job.progress.percent
 
-# retrieve the current print job information
-Octoprint::Job.current
+# Upload and start a print job
+Octoprint::Files.upload("awesome_model.gcode", "local")
+Octoprint::Files.select("local", "awesome_model.gcode") 
+Octoprint::Job.start
 ```
 
 To generate an app key on your Octoprint's instance, log in to it, click on the Settings button in the top menu and then go to the "Application Keys".
 
+## 🛡️ Type Safety with Sorbet
+
+This gem provides complete type safety with Sorbet integration:
+
+```ruby
+# All models are fully typed
+job = Octoprint::Job.current
+job.file.name         # => String (typed)
+job.progress.percent  # => Float | nil (typed)
+
+# IDE autocomplete and compile-time error checking
```

Comment:
Maybe specify that it only works with IDEs with proper config, like VSCode with ruby-lsp and sorbet plugins